### PR TITLE
[Chore] Make S3 replicas as optional variables in stack/app

### PR DESCRIPTION
**Pull request by samkahchiin**
### Description
Make S3 replicas as optional variables in stack/app

### CODE REVIEW

File: stack/app/README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ module "stack" {
       bucket_name                     = "project-staging-media",
       versioning                      = true,
       primary_storage_class_retention = 0,
-      replicas                        = []
+      # replicas is an optional variables, it can be omitted
```
Comment (swiknaba) [inline stack/app/README.md:85]:
```suggestion

      # optional      
      replicas = []
```

Comment (swiknaba) [reply stack/app/README.md:85]: to be consistent with other readmes. Also easier to read and shows the type (array).

File: stack/app/variables.tf [hunk 2]
```diff
@@ -59,11 +59,11 @@ variable "private_buckets_list" {
     bucket_name                     = string
     versioning                      = bool
     primary_storage_class_retention = number
-    replicas = list(object({
+    replicas = optional(list(object({
       bucket_arn = string
       kms_arn    = string
       region     = string
-    }))
+    })))
```
Comment (swiknaba) [inline stack/app/variables.tf:66]:
```suggestion
    })), [])
```

does this work to have a default, so we do not need to compare to null everywhere? https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes

File: stack/app/variables.tf [hunk 3]
```diff
@@ -76,9 +76,9 @@ variable "public_buckets_list" {
     bucket_name                     = string
     versioning                      = bool
     primary_storage_class_retention = number
-    replicas = list(object({
+    replicas = optional(list(object({
       bucket_arn = string
-    }))
+    })))
```
Comment (swiknaba) [inline stack/app/variables.tf:81]: same here, if that works

**Review by swiknaba — APPROVED** [review]
Thanks for updating the changelog! Left non-blocking nit picks (not sure, if a default for a non-primitive object works)

File: stack/app/variables.tf (hunk 3 above)
Comment (samkahchiin) [reply stack/app/variables.tf]: Works 🎉 
Thanks for the recommendations!

**Review by swiknaba — APPROVED** [review]
👍

**Pull request merged**